### ISSUE TITLE
Embedding padding_ix=0 is not zero across timesteps

### ISSUE THREAD

**Issue by codedecde** [pytorch/pytorch#3711]
I noticed that the padding_ix does not return a zero embedding vector after a gradient update.
I declare the embedding layer as 
`self.embedding = nn.Embedding(5, 20, padding_ix=0, sparse=True)`
And check the value of embedding of index 0 as
`e0 = self.embedding(Variable(torch.LongTensor([0])))`
I was expecting the value to be zero, but after a single gradient update, the value changes.
Doesn't padding_ix=0 ensure that index 0 always returns a 0 vector ?
Thank you for any help !
